packages/jest-diff/src/diff_strings.js

```diff
@@ -34,45 +36,120 @@ type Hunk = {|
   oldStart: number,
 |};
 
-const getColor = (added: boolean, removed: boolean) =>
-  added ? chalk.red : removed ? chalk.green : chalk.dim;
+type DIFF_D = -1 | 1 | 0; // diff digit: removed | added | equal
+
+// Given chunk, return diff character.
+const getC = (chunk): string => (chunk.removed ? '-' : chunk.added ? '+' : ' ');
+
+// Given diff character by getC from chunk or line from hunk, return diff digit.
+const getD = (c: string): DIFF_D => (c === '-' ? -1 : c === '+' ? 1 : 0);
 
-const getBgColor = (added: boolean, removed: boolean) =>
-  added ? chalk.bgRed : removed ? chalk.bgGreen : chalk.dim;
+// Color for text of line.
+// If compared lines are equal and expected and received are data structures,
+// then delta is difference in length of original lines.
+const getColor = (d: DIFF_D, delta?: number) =>
+  d === 1 ? chalk.red : d === -1 ? chalk.green : delta ? chalk.cyan : chalk.dim;
```

Comment:
This is really hard to read. Would it hurt to use one if-statement and one ternary? :D